Source: me@gmail.com
Upstream: ugly patch, not yet
Reason: Add missing icon field to desktop entry. Add enable_{alsa,oss} options.
Index: res/fqterm.desktop
===================================================================
--- res/fqterm.desktop	(版本 790)
+++ res/fqterm.desktop	(工作副本)
@@ -3,6 +3,7 @@
 Type=Application
 Name=FQTerm
 Exec=fqterm -caption "%c" %i %u
+Icon=/usr/share/FQTerm/pic/fqterm.png
 GenericName=BBS Client
 GenericName[zh_CN]=BBS 客户端程序
 GenericName[zh_TW]=BBS 客戶端程式
Index: CMakeLists.txt.linux
===================================================================
--- CMakeLists.txt.linux	(版本 790)
+++ CMakeLists.txt.linux	(工作副本)
@@ -1,24 +1,32 @@
+OPTION(ENABLE_ALSA "Add alsa support to play wav files" ON)
+OPTION(ENABLE_OSS "Add oss support to play wav files" OFF)
+
 find_package(OpenSSL REQUIRED)
-find_package(ALSA)
-find_package(OSS)
+if (ENABLE_ALSA)
+  find_package(ALSA)
+endif (ENABLE_ALSA)
 
+if (ENABLE_OSS)
+  find_package(OSS)
+endif (ENABLE_OSS)
+
 set(CMAKE_VERBOSE_MAKEFILE ON)
 
 if (FQTERM_USE_STATIC_QT)
   ADD_DEFINITIONS(-DFQTERM_USE_STATIC_QT)
 endif(FQTERM_USE_STATIC_QT)
 
-if (ALSA_FOUND)
+if (ENABLE_ALSA AND ALSA_FOUND)
   ADD_DEFINITIONS(-DAUDIO_ALSA)
   SET(AUDIO_INCLUDE_DIR "${ALSA_INCLUDE_DIR}")
   SET(AUDIO_LIBRARY "${ALSA_LIBRARY}")
-else (ALSA_FOUND)
-  if (OSS_FOUND)
-	ADD_DEFINITIONS(-DAUDIO_OSS)
-	SET(AUDIO_INCLUDE_DIR "${OSS_INCLUDE_DIR}")
-	SET(AUDIO_LIBRARY)
-  endif (OSS_FOUND)
-endif(ALSA_FOUND)
+else (ENABLE_ALSA AND ALSA_FOUND)
+  if (ENABLE_OSS AND OSS_FOUND)
+    ADD_DEFINITIONS(-DAUDIO_OSS)
+    SET(AUDIO_INCLUDE_DIR "${OSS_INCLUDE_DIR}")
+    SET(AUDIO_LIBRARY)
+  endif (ENABLE_OSS AND OSS_FOUND)
+endif (ENABLE_ALSA AND ALSA_FOUND)
 
 ADD_DEFINITIONS(-Wall)
 ADD_DEFINITIONS(-D_OS_LINUX_)
@@ -49,12 +57,12 @@
 
 if (FQTERM_USE_STATIC_QT)
   set(FQ_QT_STATIC_PLUGINS
-  ${QT_PLUGINS_DIR}/imageformats/libqgif.a    
-  ${QT_PLUGINS_DIR}/imageformats/libqjpeg.a    
-  ${QT_PLUGINS_DIR}/imageformats/libqmng.a    
-  ${QT_PLUGINS_DIR}/codecs/libqcncodecs.a   
-  ${QT_PLUGINS_DIR}/codecs/libqjpcodecs.a   
-  ${QT_PLUGINS_DIR}/codecs/libqkrcodecs.a   
+  ${QT_PLUGINS_DIR}/imageformats/libqgif.a
+  ${QT_PLUGINS_DIR}/imageformats/libqjpeg.a
+  ${QT_PLUGINS_DIR}/imageformats/libqmng.a
+  ${QT_PLUGINS_DIR}/codecs/libqcncodecs.a
+  ${QT_PLUGINS_DIR}/codecs/libqjpcodecs.a
+  ${QT_PLUGINS_DIR}/codecs/libqkrcodecs.a
   ${QT_PLUGINS_DIR}/codecs/libqtwcodecs.a
   jpeg
   )
@@ -62,13 +70,13 @@
 
 target_link_libraries(fqterm.bin
   fqterm_main
-  fqterm_terminal 
-  fqterm_protocol 
-  fqterm_utilities 
+  fqterm_terminal
+  fqterm_protocol
+  fqterm_utilities
   fqterm_ui
-  fqterm_common 
+  fqterm_common
   ${QT_QTNETWORK_LIBRARIES}
-  ${QT_QTSCRIPT_LIBRARIES}  
+  ${QT_QTSCRIPT_LIBRARIES}
   ${FQ_QT_STATIC_PLUGINS}
   ${QT_LIBRARIES}
   ${OPENSSL_LIBRARIES}
